### PR TITLE
feat: add tests for transaction encoding and decoding

### PR DESCRIPTION
**Pull request by dlealmac**
## Description
Adds dedicated unit coverage for the transaction encoder path (Type 0 / Type 1 / Type 2 / Type 4 and TransactionEncoderFactory), plus cross-cutting checks that pin wire and signing bytes against an independent RLP reference and exercise encode→parse→re-encode round trips.

## Motivation and Context
Complements existing encoder and typed-transaction coverage with dedicated unit tests for Type0 / Type1 / Type2 and factory routing, plus round-trip and independent golden checks so encodeForSigning / encodeSigned stay aligned with the RSKIP wire format (RSKIP-543 envelope; RSKIP-546 type 1/2; RSKIP-545 type 4; legacy EIP-155).

RSK-namespace Type2RSK is out of scope for this branch since it was handled by another PR.

## How Has This Been Tested?
New tests coverage includes chainId 0 and 33 for legacy, chainId 33 and 200 (128–255 unsigned-byte path) for typed transactions, signing-payload type-prefix checks, and golden byte equality against ReferenceRlpEncoder.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

### CODE REVIEW

File: rskj-core/src/test/java/org/ethereum/core/transaction/encoder/ReferenceRlpEncoder.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+ * <p>This is a deliberate second implementation written straight from the specs. It MUST NOT
+ * delegate to production code ({@code org.ethereum.util.RLP} or the {@code *TransactionEncoder}s):
+ * its whole purpose is to fail if the production encoder drifts from the bytes other compliant
+ * clients would accept. A round-trip test cannot catch that, because encoder and parser can drift
+ * together and stay self-consistent.
+ *
+ * <p>The field values mirror {@link EncoderTestSupport}; if you change a fixture there, change it
+ * here too - do not make this class read from the production encoders to "sync" them.
+ */
+final class ReferenceRlpEncoder {
```
Comment (meldsun0) [inline rskj-core/src/test/java/org/ethereum/core/transaction/encoder/ReferenceRlpEncoder.java:40]: Generate the golden vectors once with ReferenceRlpEncoder, then store the resulting hexadecimal values as immutable constants. This makes the expected outputs independent of the reference implementation and ensures the tests fail if either the production encoder or the reference encoder accidentally changes.

File: rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * <p>Round-trip tests stay green when the encoder and parser drift from the spec together;
+ * this test is the only one that fails in that case. If it fails, the encoder output no
+ * longer matches what other compliant clients would produce - fix the encoder, do not
+ * "sync" {@link ReferenceRlpEncoder} to the production output.
+ */
+class GoldenTransactionEncoderTest {
+
+    private static final Map<String, String[]> VECTORS = ReferenceRlpEncoder.goldenVectors();
+
+    private static Stream<Arguments> goldenCases() {
```
Comment (meldsun0) [inline rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java:56]: Consider extending the golden vector coverage to include the remaining edge-case chainId combinations, particularly:

- type1-chain0
- type1-chain200
- type2-chain0
- type4-chain0 with both chainId: transaction.chainId and authorization.chainId
- type4-chain200 with both chainId: transaction.chainId and authorization.chainId
- legacy-chain200

The current suite covers the common paths, but these additional cases would exercise canonical encoding of chainId = 0, unsigned-byte handling for 128–255, and multi-byte EIP-155 v encoding for legacy transactions (chainId = 200), providing more complete interoperability coverage.

File: rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        return Stream.of(
+                Arguments.of("legacy-chain0", (Supplier<Transaction>) () -> unsignedLegacy((byte) 0), FIXED_V),
+                Arguments.of("legacy-chain33", (Supplier<Transaction>) () -> unsignedLegacy(CHAIN_ID), FIXED_V),
+                Arguments.of("type1-chain33", (Supplier<Transaction>) EncoderTestSupport::unsignedType1, FIXED_V),
+                Arguments.of("type2-chain33", (Supplier<Transaction>) EncoderTestSupport::unsignedType2, FIXED_V),
+                Arguments.of("type2-chain33-yParity0", (Supplier<Transaction>) EncoderTestSupport::unsignedType2, FIXED_V_Y_PARITY_0),
+                Arguments.of("type2-chain200", (Supplier<Transaction>) () -> EncoderTestSupport.unsignedType2(HIGH_CHAIN_ID), FIXED_V),
+                Arguments.of("type4-chain33", (Supplier<Transaction>) EncoderTestSupport::unsignedType4, FIXED_V)
```
Comment (italo-sampaio) [inline rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java]: **Vary the field shape as well as the chainId.**

Every vector here uses one shape: a 20 byte `to`, empty `data`, and an empty access list. Three cases get no coverage.

Contract creation. `RLP.encodeRskAddress` returns `encodeElement(null)` for the null address, a separate branch from the 20 byte case. No vector in this file reaches it, and no other test in this PR does either.

Non-empty `data`. Every item in every current vector is shorter than 56 bytes, so `ReferenceRlpEncoder.encodeLength` only ever takes its short form for items. The long form runs for the outer list and never for an item. A `data` field of 56 bytes or more reaches it. That branch carries no coverage today, inside the class every assertion in this file depends on.

Non-empty access list. `Type1TransactionEncoderTest.encodeForSigning_changesWhenAccessListChanges` shows that the payload changes and stops there. A wrong nesting or a wrong item order still passes. A vector with one address and one storage key pins the layout.

A separate comment here asks for more chainId values. These three are a different axis and fit in the same pass.

**Review by italo-sampaio — APPROVED** [review]
Left one suggestion, approving regardless.

File: rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java (hunk 3 above)
Comment (dlealmac) [reply rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java]: done

File: rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java (hunk 2 above)
Comment (dlealmac) [reply rskj-core/src/test/java/org/ethereum/core/transaction/encoder/GoldenTransactionEncoderTest.java:56]: done

File: rskj-core/src/test/java/org/ethereum/core/transaction/encoder/ReferenceRlpEncoder.java (hunk 1 above)
Comment (dlealmac) [reply rskj-core/src/test/java/org/ethereum/core/transaction/encoder/ReferenceRlpEncoder.java:40]: done

**Review by italo-sampaio — APPROVED** [review]

**Review by meldsun0 — APPROVED** [review]
